packages/lisk-p2p/src/p2p.ts

```diff
@@ -344,17 +344,13 @@ export class P2P extends EventEmitter {
 
 				const wsPort: number = parseInt(queryObject.wsPort, BASE_10_RADIX);
 				const peerId = constructPeerId(socket.remoteAddress, wsPort);
-				const queryOptions =
-					typeof queryObject.options === 'string'
-						? JSON.parse(queryObject.options)
-						: undefined;
 
 				const incomingPeerInfo: P2PDiscoveredPeerInfo = {
+					...queryObject,
```

Comment:
would this work in case of nested options? im not sure nested object is supported though
ie)
`ws://some-url?httpPort=3000&options={"id": "some-id"}`

Reply:
Yes, it will work with nested objects as well, in the test cases we are also passing options and it still shows up in incoming peer info.

Reply:
As discussed, we are going to add the options fields to wrap all the custom fields. This issue will now be covered in PR #1170